src/main/java/com/petfinder/service/UserService.java

```diff
@@ -38,16 +42,75 @@ public void register(String login, String password, String repeatPassword,
                             "Passwords does not match."
                     );
                 }
-            }
-        }
-    }
-
-    public boolean checkIfUserIsLogged() {
-        if (SecurityContextHolder.getContext().getAuthentication() instanceof AnonymousAuthenticationToken) {
-            return false;
-        }
-        return true;
-    }
+			}
+		}
+	}
+		
+	@Transactional
+	public void changeUserData(String currentPassword, String newPassword, String repeatPassword, String newEmail, String repeatEmail) throws PasswordsDoesNotMatchException, EmailExistsException, EmailsDoesNotMatchException, InvalidUserPasswordException{
+		if(checkIfUserIsLogged()){
+	        String login = SecurityContextHolder.getContext().getAuthentication().getName();
+			User user = userRepository.findOneByLogin(login);
+			if(this.passwordEncoder.matches(currentPassword, user.getPassword())){
+				boolean isPasswordUpdateRequired = passwordNeedsToBeUpdated(newPassword,repeatPassword);
+				boolean isEmailUpdateRequired = emailNeedsToBeUpdated(newEmail,repeatEmail);
+				
+				if(isPasswordUpdateRequired){
+					updatePassword(user,newPassword, repeatPassword);
+				}
+				if(isEmailUpdateRequired){
+					updateEmail(user,newEmail, repeatEmail);
+				}
+				
+				if(isPasswordUpdateRequired == true || isEmailUpdateRequired == true){
+					userRepository.save(user);
+				}
+			} else{
+				throw new InvalidUserPasswordException("User's password is invalid.");
+			}
+		}
+	}
+	
+	private void updateEmail(User user, String newEmail, String repeatEmail) throws EmailExistsException, EmailsDoesNotMatchException{
+		if(newEmail.equals(repeatEmail)){
+			if(verifyEmail(newEmail)){
+				user.setEmail(newEmail);
+			}
+		}	else{
+			throw new EmailsDoesNotMatchException("Given emails does not match.");
+		}
+	}
+	
+	private void updatePassword(User user, String newPassword, String repeatPassword) throws PasswordsDoesNotMatchException{
+		if(newPassword.equals(repeatPassword)){
+			user.setPassword(hashPassword(newPassword));
+		} else{
+			throw new PasswordsDoesNotMatchException("Given passwords does not match.");
+		}
+	}
+	
+	private boolean passwordNeedsToBeUpdated(String newPassword, String repeatPassword){
+		if((newPassword.equals("")||newPassword == null)&&(repeatPassword.equals("")||repeatPassword == null)){
+			return false;
+		}
+		return true;
+	}
+	
+	private boolean emailNeedsToBeUpdated(String newEmail, String repeatEmail){
+		if((newEmail.equals("")||newEmail == null)&&(repeatEmail.equals("")||repeatEmail == null)){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The null check here is misplaced; if the variable is null there will be a NullPointerException](https://www.codacy.com/app/matikz1110/PetFinder/file/2655793372/issues/source?bid=3239464&fileBranchId=3327196#l100)